src/app/service/game-control.service.ts

```diff
@@ -49,155 +45,167 @@ export class GameControlService {
     [3, 5, 7],
   ];
 
-  private gameBoard: BehaviorSubject<number[]> = new BehaviorSubject<number[]>(
-    new Array(9)
-  );
-  private games$$: BehaviorSubject<gameStream> =
-    new BehaviorSubject<gameStream>(new ReplaySubject<move>());
-
-  private score: BehaviorSubject<Score> = new BehaviorSubject<Score>({
-    cross: 0,
-    circle: 0,
-  });
-  private activePlayer: BehaviorSubject<string> = new BehaviorSubject<string>(
-    this.togglePlayer()
-  );
-  private winner: BehaviorSubject<string> = new BehaviorSubject<string>('');
-  private round: BehaviorSubject<number> = new BehaviorSubject<number>(1);
-  private gameActive: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(
-    true
-  );
-
-  constructor() {}
-
-  getGameBoard() {
-    return this.gameBoard.asObservable();
+  constructor(private firestore: Firestore, private router: Router) {}
+
+  getAllGames() {
+    return collectionData(collection(this.firestore, 'game'), {
+      idField: 'gameId',
+    });
   }
 
-  getGameStats() {
-    return combineLatest([
-      this.getWinner(),
-      this.getScore('circle'),
-      this.getScore('cross'),
-      this.round.asObservable(),
-      this.activePlayer.asObservable(),
-      this.getGameStatus(),
-    ]).pipe(
-      map(([winner, circle, cross, round, player, active]) => ({
-        winner,
-        circle,
-        cross,
-        round,
-        player,
-        active,
-      }))
-    );
+  getStats() {
+    return this.gameData$;
   }
 
-  getAllMoves() {
-    return this.games$$.pipe(
-      switchMap((game$) =>
-        game$.pipe(
-          scan((acc: move[], curr: move) => (curr ? [...acc, curr] : []), [])
-        )
-      ),
-      shareReplay(1)
-    );
+  getBoard() {
+    return this.gameData$.pipe(map(({ board }) => board));
   }
 
-  getSingleMove(field: number) {
-    return this.getAllMoves().pipe(
-      filter((moves) => moves.some((move) => move.field === field)),
-      withLatestFrom(this.gameActive),
-      tap(([moves, active]) => {
-        if (active) this.findWinner(moves, this.returnPlayer(moves, field));
-      }),
-      map(([moves, _]) => this.returnPlayer(moves, field))
-    );
+  createNewGame(username: string) {
+    const id = String(Date.now());
+    this.gameId = id;
+    this.player = 'cross';
+    setDoc(doc(this.firestore, 'game', id), {
```

Comment:
I would extract all communication with Firebase to the fasade. So we will simplify this service - and only fasade will be communicating with Firebase.

Reply:
done